# 주의
- 모든 코드는 위의 섹션부터 순서대로 실행해야 정상 작동합니다.

In [ ]:
# 환경 설정
import pandas as pd

In [13]:
# 데이터셋을 불러와 head(), info(), describe() 결과를 출력합니다.
# 반드시 data/ 경로에 'heart_disease.csv' 파일이 존재해야 합니다.
# 없는 경우에는 다음을 실행해 생성: python data/data.py

# 파일 불러오기
df = pd.read_csv('../data/heart_disease.csv')

# head()
print(df.head())

# info()
print(df.info(verbose=True, show_counts=True))

# describe()
print(df.describe())

   age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   1       145   233    1        2      150      0      2.3      3   
1   67    1   4       160   286    0        2      108      1      1.5      2   
2   67    1   4       120   229    0        2      129      1      2.6      2   
3   37    1   3       130   250    0        0      187      0      3.5      3   
4   41    0   2       130   204    0        2      172      0      1.4      1   

    ca  thal  target  
0  0.0   6.0       0  
1  3.0   3.0       1  
2  2.0   7.0       1  
3  0.0   3.0       0  
4  0.0   3.0       0  
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       303 non-null    int64  
 1   sex       303 non-null    int64  
 2   cp        303 non-null    int64  
 3   trestbps  303 non-null    int64  
 4   chol      303 non-null    int64

In [18]:
# 타깃 클래스 분포 출력
print(df['target'].value_counts(normalize=True))

target
0    0.541254
1    0.458746
Name: proportion, dtype: float64


# 타깃 클래스 분포가 평가 지표 선택에 미치는 영향
- 타깃 클래스는 0(정상)과 1(심장병)으로 분류되며, 그 분포는 0이 약 0.54, 1이 약 0.46으로 측정되었다. 정상과 심장병 비율이 각각 50%에 가까워 적절히 균형 잡혀 있어 모델 학습 시 데이터 쏠림 현상이 발생하지 않아 정확도를 신뢰할 수 있다.
- 심장병 진단이라는 목적상 실제 심장병 환자를 정상으로 잘못 예측(False Negative)했을 때 환자가 매우 위험할 수 있으므로, 최종적으로 모델을 평가할 때는 재현율을 평가 지표로 함께 고려해야 한다.